Common/Scheduling/DateRules.cs

```diff
@@ -122,87 +123,144 @@ public IDateRule EveryDay(Symbol symbol)
         }
 
         /// <summary>
-        /// Specifies an event should fire on the first of each month
+        /// Specifies an event should fire on the first of each month + offset
         /// </summary>
-        /// <returns>A date rule that fires on the first of each month</returns>
-        public IDateRule MonthStart()
+        /// <param name="daysOffset"> The amount of days to offset the schedule by; must be between 0 and 30.</param>
+        /// <returns>A date rule that fires on the first of each month + offset</returns>
+        public IDateRule MonthStart(int daysOffset = 0)
         {
-            return new FuncDateRule("MonthStart", (start, end) => MonthStartIterator(null, start, end));
+            return MonthStart(null, daysOffset);
         }
 
         /// <summary>
-        /// Specifies an event should fire on the first tradeable date for the specified
-        /// symbol of each month
+        /// Specifies an event should fire on the first tradable date + offset for the specified symbol of each month
         /// </summary>
-        /// <param name="symbol">The symbol whose exchange is used to determine the first
-        /// tradeable date of the month</param>
-        /// <returns>A date rule that fires on the first tradeable date for the specified security each month</returns>
-        public IDateRule MonthStart(Symbol symbol)
+        /// <param name="symbol">The symbol whose exchange is used to determine the first tradable date of the month</param>
+        /// <param name="daysOffset"> The amount of days to offset the schedule by; must be between 0 and 30</param>
```

Comment:
nit - think would be nice to add `The amount of **tradable** days` which makes it different from the other method overload, same for the other methods